addons/web/models/base_document_layout.py

```diff
@@ -33,9 +45,27 @@ class BaseDocumentLayout(models.TransientModel):
     logo_secondary_color = fields.Char(compute="_compute_logo_colors")
 
     report_layout_id = fields.Many2one('report.layout')
-    preview = fields.Html(compute='_compute_preview')
 
-    @api.depends('logo_primary_color', 'logo_secondary_color', 'primary_color', 'secondary_color')
+    # All the sanitization get disabled as we want true raw html to be passed to an iframe.
+    preview = fields.Html(compute='_compute_preview',
+                          sanitize=False,
+                          sanitize_tags=False,
+                          sanitize_attributes=False,
+                          sanitize_style=False,
+                          sanitize_form=False,
+                          strip_style=False,
+                          strip_classes=False)
+
+    # Those following fields are required as a company to create invoice report
+    partner_id = fields.Many2one(related='company_id.partner_id')
```

Comment:
fields in readonly

Reply:
@kebeclibre And meant to be ? They are only used to retrieve info that could be required in the invoice template, they are never changes from the wizard.

Reply:
yes I know, It is both for clarity in the code and some sense of security... One should never be able to change those data from a simple wizard

Reply:
oh you're talking about the preview field, I though you were talking about the related fields below.

Reply:
héhéhé, nope, I'm definitely talking about the related fields (partner, vat etc.....)
Their presence in code is just to provide a res.company-like API, they should never be writable from the wizard

Reply:
I though related field were readonly by default ? You're saying you want it to be more explicit ?